stores/sql.go

```diff
@@ -175,13 +175,16 @@ func NewSQLStore(conn, connMetrics gorm.Dialector, alerts alerts.Alerter, partia
 		}
 	}
 
-	// Check if any indices are missing after migrations.
+	// Fix missing indices.
 	detectMissingIndices(db, func(dst interface{}, name string) {
 		t := reflect.TypeOf(dst)
 		if t.Kind() == reflect.Ptr {
 			t = t.Elem()
 		}
-		logger.Warnw("missing index", "table", t.Name(), "field", name)
+		logger.Infow("creating missing index", "table", t.Name(), "field", name)
+		if err := db.Migrator().CreateIndex(dst, name); err != nil {
```

Comment:
I wouldn't do that that personally but fine if you want to add that. Probably best to rename it to `tryEnsureIndices` to make sure the caller knows it's going to try and create them when it's called.